### PR TITLE
Feature request - specify branch/sha/tag when adding new projects

### PR DESCRIPTION
**Pull request by ktmills**
Fixes # [mateodelnorte/meta#128](https://github.com/mateodelnorte/meta/issues/128)

Added child repo branch optional argument with adding new projects

### CODE REVIEW

**Comment by niedfelj** [comment]
Ideally this could be specified on a per-project basis, looks like this is for every repo in the meta to track on the same branch? We often have stable branches that don't need any changes building or depending on other feature branches, where it would be useful to have meta track across

File: bin/meta-project-add [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env node
 
 const usage = () => {
-  console.log('Usage: meta-project-add <destFolder> <childRepoUrl>');
+  console.log('Usage: meta-project-add <destFolder> <childRepoUrl> [childRepoBranch]');
```
Comment (mateodelnorte) [inline bin/meta-project-add]: can we call this defaultBranch?

File: bin/meta-project-add [hunk 2]
```diff
@@ -10,6 +10,8 @@ const { argv } = process;
 
 const destArg = argv[2] === 'blank' ? argv[3] : argv[2];
 const repoUrl = argv[3] === 'blank' ? argv[4] : argv[3];
+const repoBranch = argv[4] === 'blank' ? (argv[5] === 'blank' ? 'master' : argv[5]) : argv[4];
```
Comment (mateodelnorte) [inline bin/meta-project-add]: can we name this defaultBranch, as well?

**Comment by mateodelnorte** [comment]
@niedfelj not quite sure what you mean. Can you rephrase? @ktmills's changes appear to be per-project. Note, these are just for *default* branches to clone down.

**Comment by niedfelj** [comment]
@mateodelnorte I see now, you are correct. I thought "projects" had separate keys in it where it would make sense to track branches/commits in there, but he's doing the same in a separate branches config.

**Comment by mateodelnorte** [comment]
@ktmills, @patrickleet has revamped the meta-project commands in response to a lot of feedback we've gotten. there is now `meta project import` and `meta project create`, and we've deprecated `meta project add`. `meta project create` does what `meta project add` used to do, but no longer pulls from the remote after adding the folder. 

if you want to add an existing project, `meta project import` is where this logic will go – but I do still have overall concerns about whether this is needed, as the default branch (as selected in github) should always be checked out by default.
